state-sync/storage-service/server/src/utils.rs

```diff
@@ -175,3 +177,48 @@ pub fn notify_peer_of_new_data<T: StorageReaderInterface>(
 
     Ok(transformed_data_response)
 }
+
+/// An utility that calls and times the given function. The metric histogram
+/// is updated with the given labels (e.g., peer, request result and duration).
+/// If no start time is specified, the timer begins before calling the function.
+pub fn execute_and_time_duration<T, F>(
```

Comment:
Seems similar in spirit (but not exact functionality) to `monitor!` in consensus. Maybe there's some unifying to do down the road?

Reply:
Yeah, definitely. I think storage also does its own thing here, too 😄